.github/workflows/codeql.yml

```diff
@@ -0,0 +1,56 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+name: CodeQL Security Scan
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'src/**'
+  push:
+    branches:
+      - main
```

Comment:
The dependency review throws an error if you include `on.push` in the configuration, but this is _required_ for the CodeQL action. Therefore we need two different workflows for these jobs.

[This example](https://github.com/stephenfuqua/Suite-3-Performance-Testing/actions/runs/2386742890) demonstrates the problem.